plugins/inputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -82,14 +86,11 @@ func (ps *PubSub) SetParser(parser parsers.Parser) {
 // Two goroutines are started - one pulling for the subscription, one
 // receiving delivery notifications from the accumulator.
 func (ps *PubSub) Start(ac telegraf.Accumulator) error {
-	if ps.Subscription == "" {
-		return fmt.Errorf(`"subscription" is required`)
-	}
-
-	if ps.Project == "" {
-		return fmt.Errorf(`"project" is required`)
+	var err error
+	ps.decoder, err = internal.NewContentDecoder(ps.ContentEncoding)
```

Comment:
is it worth checking the encoding type and if it is identity skipping this?

Reply:
I'm fine either way. Please note that if we only do the decoding in "non-identify" cases we do not need the lock to be acquired probably keeping up the performance for the non-compressed case.

Reply:
Feels a lil error prone if we don't set a default decoder. Empty string for ContentEncoding creates a 'identity' decoder. So it seemed safe.

Reply:
is a "non-identify" case, cases where the user doesn't choose "identity" as the content encoder? As the "identity" content encoding is still a valid encoder type.

Reply:
I updated to assume the default value is "identity"